commit ef07c2af665e76e66b69fa216b890ba4c88b0b01
Author: John Levon <john.levon@joyent.com>
Date:   2019-01-25T17:15:35+00:00 (9 months ago)
    
    TRITON-1158 triton docs build needs fixes on SmartOS
    TRITON-1017 iPXE should be provided directly from joyent/ipxe

diff --git a/Makefile b/Makefile
index 8545cbc..7375b68 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -18,6 +18,7 @@ RAMSEY = node_modules/.bin/ramsey
 MD_FILES = \
 	docs/developer-guide/repos.md
 
+CLEAN_FILES += build/repos.json
 
 
 include ./tools/mk/Makefile.defs
@@ -36,13 +37,13 @@ docs: $(MD_FILES)
 docs/developer-guide/repos.md: $(RAMSEY) docs/developer-guide/repos.md.in build/repos.json
 	$(RAMSEY) -d build -j repos.json "$@.in" "$@"
 
+JSON_SCRIPT = 'var self = this; \
+	this.name = /([^:/]+).git$$/.exec(this.git)[1]; \
+	this.url = "https://github.com/joyent/" + this.name; \
+	(this.tags || ["none"]).forEach(function (t) { self[t] = true; });'
+
 build/repos.json: build etc/repos.json
-	$(JSON) -f ./etc/repos.json -e ' \
-	    var self = this; \
-	    this.name = /([^:/]+).git$$/.exec(this.git)[1]; \
-	    this.url = "https://github.com/joyent/" + this.name; \
-	    (this.tags || ["none"]).forEach(function (t) { self[t] = true; }); \
-	    ' > "$@"
+	$(JSON) -f ./etc/repos.json -e $(JSON_SCRIPT) > "$@"
 
 build:
 	mkdir -p "$@"
diff --git a/docs/developer-guide/repos.md b/docs/developer-guide/repos.md
index 19864c9..f6c5a69 100644
--- a/docs/developer-guide/repos.md
+++ b/docs/developer-guide/repos.md
@@ -107,10 +107,10 @@ the OS image, and form the foundation for SDC:
 Build tools are used for creating the zone images:
 
 * [sdcnode](https://github.com/joyent/sdcnode): Tools for creation of prebuilt node tarballs for SDC components.
-* [sdcboot](https://github.com/joyent/sdcboot): SDC FDUM environment
 * [sdc-headnode](https://github.com/joyent/sdc-headnode): Repository for building headnode images for SDC, and the intial setup and configuration of the headnode itself
 * [sdc-scripts](https://github.com/joyent/sdc-scripts): Common scripts for configuring and setting up SDC zones.
 * [mountain-gorilla](https://github.com/joyent/mountain-gorilla): Builder of all the SDC bits.
+* [ipxe](https://github.com/joyent/ipxe): SDC iPXE implementation
 
 
 There are also services responsible for syncing data between datacenters or
diff --git a/docs/developer-guide/repos.md.in b/docs/developer-guide/repos.md.in
index 8c2dbed..9e8693d 100644
--- a/docs/developer-guide/repos.md.in
+++ b/docs/developer-guide/repos.md.in
@@ -1,7 +1,7 @@
-# SmartDataCenter repos
+# Triton DataCenter repos
 
 This page holds an overview of the source repositories that make up
-SmartDataCenter (SDC). See also [the reference](../reference.md).
+Triton DataCenter. See also [the reference](../reference.md).
 
 
 The customer-facing self-service API:
diff --git a/etc/repos.json b/etc/repos.json
index 44d6b95..8cf37d8 100644
--- a/etc/repos.json
+++ b/etc/repos.json
@@ -253,13 +253,6 @@
       "cli"
     ]
   },
-  {
-    "git": "git@github.com:joyent/sdcboot.git",
-    "desc": "SDC FDUM environment",
-    "tags": [
-      "build"
-    ]
-  },
   {
     "git": "git@github.com:joyent/sdc-ufds.git",
     "desc": "SDC LDAP Server",
@@ -509,6 +502,13 @@
       "build"
     ]
   },
+  {
+    "git": "git@github.com:joyent/ipxe.git",
+    "desc": "SDC iPXE implementation",
+    "tags": [
+      "build"
+    ]
+  },
   {
     "git": "git@github.com:joyent/eng.git",
     "desc": "Joyent Engineering Guide",
diff --git a/repos.json b/repos.json
index 9b91410..4f7faad 100644
--- a/repos.json
+++ b/repos.json
@@ -83,6 +83,13 @@
         {
             "name": "grafana"
         },
+        {
+            "name": "ipxe",
+            "labels": {
+                "release": true,
+                "mg": "ipxe"
+            }
+        },
         {
             "name": "java-triton"
         },
@@ -1045,7 +1052,6 @@
         "image-converter",
         "imagetools",
         "innovator-workshop",
-        "ipxe",
         "java-fast-md5",
         "java-http-signature",
         "java-manta",
